### PR TITLE
Show page spinner on resources and environments dropdown until the ajax finishes (#3187)

### PR DESCRIPTION
**Pull request by GaneshSPatil**
screen shot 2017-03-17 at 10 41 32 am
screen shot 2017-03-17 at 10 42 09 am

### CODE REVIEW

**Comment by ketan** [comment]
Looks good, can we handle errors in case of ajax failures (or timeouts)?

File: server/webapp/WEB-INF/rails.new/webpack/views/agents/environments_list_widget.js.msx [hunk 1]
```diff
@@ -47,10 +47,19 @@ const EnvironmentsListWidget = {
   },
 
   view (vnode) {
-
     const ctrl = vnode.state;
     const args = vnode.attrs;
 
+    const spinner = (
+      <div class="agent-button-group-dropdown env-dropdown empty-container" onclick={ctrl.closeDropdown}>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/webpack/views/agents/environments_list_widget.js.msx]: These are the same classes as the ones defined in `environmentDropDownClass`. Dedup it? `class` attribute accepts arrays (even with null and undefined objects), and they usually work better with optional class names.

File: server/webapp/WEB-INF/rails.new/webpack/views/agents/resources_list_widget.js.msx [hunk 2]
```diff
@@ -76,8 +76,18 @@ const ResourcesListWidget = {
   },
 
   view (vnode) {
+    const spinner = (
+      <div class="agent-button-group-dropdown resource-dropdown empty-container" onclick={vnode.state.closeDropdown}>
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/webpack/views/agents/resources_list_widget.js.msx]: Same comment as above.

**Comment by GaneshSPatil** [comment]
screen shot 2017-03-29 at 11 07 02 am
screen shot 2017-03-29 at 11 08 07 am

**Pull request merged**